scripts/imgtool/image.py

```diff
@@ -255,16 +259,50 @@ def ecies_p256_hkdf(self, enckey, plainkey):
             format=PublicFormat.UncompressedPoint)
         return cipherkey, ciphermac, pubk
 
-    def create(self, key, enckey, dependencies=None):
+    def create(self, key, enckey, dependencies=None, sw_type=None):
         self.enckey = enckey
 
+        # Calculate the hash of the public key
+        if key is not None:
+            pub = key.get_public_bytes()
+            sha = hashlib.sha256()
+            sha.update(pub)
+            pubbytes = sha.digest()
+        else:
+            pubbytes = bytes(SHA256_DIGEST_SIZE)
```

Comment:
Could get the digest directly from `hashlib` with `bytes(hashlib.sha256().digest_size)`